go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -126,19 +136,23 @@ func newNodeTracker(nodeInformer coreinformers.NodeInformer) (*nodeTracker, erro
 
 // updateNodeInfo updates the node info cache, and syncs all services
 // if it changed.
-func (nt *nodeTracker) updateNodeInfo(nodeName, switchName, routerName string, nodeIPs []string, podSubnets []*net.IPNet) {
+func (nt *nodeTracker) updateNodeInfo(nodeName, switchName, routerName, chassisID string, nodeIPs []net.IP, podSubnets []*net.IPNet) {
 	ni := nodeInfo{
 		name:              nodeName,
 		nodeIPs:           nodeIPs,
 		podSubnets:        make([]net.IPNet, 0, len(podSubnets)),
 		gatewayRouterName: routerName,
 		switchName:        switchName,
+		chassisID:         chassisID,
 	}
 	for i := range podSubnets {
 		ni.podSubnets = append(ni.podSubnets, *podSubnets[i]) // de-pointer
 	}
 
+	klog.Infof("Node %s switch + router changed, syncing services", nodeName)
+
 	nt.Lock()
+	defer nt.Unlock()
```

Comment:
There is an issue here because we are locking unlocking and in `resyncFn` as well

Reply:
Seen on the job logs:
```
2023-04-05T14:17:13.680671869Z stderr F I0405 14:17:13.680575      35 node_tracker.go:152] Node ovn-worker switch + router changed, syncing services
2023-04-05T14:17:13.680685069Z stderr F fatal error: sync: unlock of unlocked mutex
2023-04-05T14:17:13.683158193Z stderr F
2023-04-05T14:17:13.683346795Z stderr F goroutine 410 [running]:
2023-04-05T14:17:13.683805699Z stderr F runtime.throw({0x1f9c27a?, 0x0?})
2023-04-05T14:17:13.684640307Z stderr F         /opt/hostedtoolcache/go/1.18.4/x64/src/runtime/panic.go:992 +0x71 fp=0xc0009f5818 sp=0xc0009f57e8 pc=0x438491
2023-04-05T14:17:13.685163012Z stderr F sync.throw({0x1f9c27a?, 0x0?})
2023-04-05T14:17:13.685722318Z stderr F         /opt/hostedtoolcache/go/1.18.4/x64/src/runtime/panic.go:978 +0x1e fp=0xc0009f5838 sp=0xc0009f5818 pc=0x46647e
2023-04-05T14:17:13.685993121Z stderr F sync.(*Mutex).unlockSlow(0xc001633c20, 0xffffffff)
2023-04-05T14:17:13.686533226Z stderr F         /opt/hostedtoolcache/go/1.18.4/x64/src/sync/mutex.go:220 +0x3c fp=0xc0009f5860 sp=0xc0009f5838 pc=0x47569c
2023-04-05T14:17:13.686760228Z stderr F sync.(*Mutex).Unlock(0x0?)
2023-04-05T14:17:13.687310333Z stderr F         /opt/hostedtoolcache/go/1.18.4/x64/src/sync/mutex.go:214 +0x29 fp=0xc0009f5880 sp=0xc0009f5860 pc=0x475629
2023-04-05T14:17:13.687446035Z stderr F github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/controller/services.(*nodeTracker).updateNodeInfo.func1()
2023-04-05T14:17:13.68798494Z stderr F  /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/ovn/controller/services/node_tracker.go:155 +0x26 fp=0xc0009f5898 sp=0xc0009f5880 pc=0x1914de6

```

Reply:
Uhm.. I think this is actually because of:
```
	nt.Lock()
	defer nt.Unlock()
	if existing, ok := nt.nodes[nodeName]; ok {
		if reflect.DeepEqual(existing, ni) {
			nt.Unlock()   // I forgot to remove this one..
			return
		}
	}
```
`resyncFn` doesn't lock.  I'll fix the former soon.

Reply:
ah yeah, I confused that with `removeNodeWithServiceReSync` that doesn't play a role here.